ts/src/main/scala/ch/epfl/ts/component/fetch/HistDataCSVFetcher.scala

```diff
@@ -79,25 +92,45 @@ class HistDataCSVFetcher(dataDir: String, currencyPair: String,
   val timer = new Timer()
   timer.schedule(new SendQuotes, 0)
   class SendQuotes extends java.util.TimerTask {
+    
     def run() {
-      if (quoteIndex < allQuotes.length) {
-        // Update the iterator variables
-        var currQ = allQuotes( quoteIndex )
-        var nextQ = allQuotes( List(quoteIndex + 1, allQuotes.length).min )
-        quoteIndex = quoteIndex + 1;
-        
-        // Send the quote and schedule next call
-        callback(currQ)
+      // Get the current quote and send it
+      callback(currQ)
+  
+      if (allQuotes.hasNext) {
+        // If there is a next quote, schedule the next call
+        currQ = nextQ
+        nextQ = allQuotes.next
         timer.schedule(new SendQuotes(), (1 / speed * (nextQ.timestamp - currQ.timestamp)).toInt)
       } else {
-        timer.cancel();
+        // Nothing more to do here, boys!
+        timer.cancel() 
       }
     }
+    
   }
   
-  // TODO
+  /**
+   * Saves the quotes this fetcher has fetched (i.e. quotes that are stored in val allQuotes)
+   * in an sqlite database of a given name.
+   * 
```

Comment:
Explicit what happens if we want to save data to a DB and also use our fetcher afterwards. Does the fetcher rewind all the way to the start after having saved the DB?